src/hint_processor/builtin_hint_processor/builtin_hint_processor_definition.rs

```diff
@@ -463,7 +494,12 @@ impl HintProcessor for BuiltinHintProcessor {
             hint_code::EC_MUL_INNER => {
                 ec_mul_inner(vm_proxy, &hint_data.ids_data, &hint_data.ap_tracking)
             }
-            code => Err(VirtualMachineError::UnknownHint(code.to_string())),
+            code => {
+                if let Some(ref executor) = self.external_executor {
+                    return executor.execute_hint(vm_proxy, exec_scopes_proxy, hint_data);
+                }
```

Comment:
What about the compilation step?

Reply:
As the external hint executor comes as an add-on to the builtin hint processor it uses the same compilation step as the builtin hint processor

Reply:
But if it's going to be handling a superset of hints then this step also needs to call a compile step for the external executor. Otherwise it will only handle built-ins.

Reply:
But that's the thing, that's true for the builtin processor only. An arbitrary external processor may have other needs. That's why we provide the ability to compile them in the first place.
For example, a Python executor that we may add would probably want to cache the compiled code instead of compiling at every call.